### DISCUSSION TITLE
Local costmap

### DISCUSSION THREAD

**Discussion by Xindictus** [gstavrinos/ai_msc_robotic_systems_assignments#45] [Q&A] [answered]
I'm trying to figure out how to update the local costmap, but, it's not that clear from the API docs. I have deduced the following:
1. **Publish to a topic**: I found out there is a topic that only has a subscriber and no publishers, which could indicate is the right one. Unfortunately I can't tell as this topic is not mentioned in the docs at all:
```shell
$ ros2 topic info /local_costmap/footprint 
Type: geometry_msgs/msg/Polygon
Publisher count: 0
Subscription count: 1
```

2. **Update using the `Costmap API`**: Using something like `setCost` could be the idea solution instead?

**Comment by gstavrinos** [comment]
You have been derailed too much. The only thing required for the solution of the exercise is to get the goal positions from the TF tree and then transform them and send them as goals to the robot. Nothing else.

I am not even sure how you ended up reading such details of the nav2 API. Just FYI, the local costmap is generally updated from sensors, which our robot has been already set up to do. You mostly need to manually modify the costmap in very special and advanced occasions.

**Reply by Xindictus** [reply]
Assignment 2 mentioned the following:
> c. Notice that the robot often collides with the table. Make the required changes in order to add the table to the local costmap.

As such, I thought I should make changes to the local costmap, since, indeed the robot collides with the table. I explain this a bit better in https://github.com/gstavrinos/ai_msc_robotic_systems_assignments/discussions/46

**Reply by gstavrinos** [reply, marked as answer]
During the lecture, we discussed that this portion of the exercise has already been solved for you, if you enable the `easy_mode` parameter of the launch file. If and only if you have the time should you run without `easy_mode` enabled.
